System.IO.Ports/SerialPort.cs

```diff
@@ -383,14 +383,14 @@ public string NewLine
         /// <exception cref="IOException">The port is in an invalid state. -or- An attempt to set the state of the underlying
         /// port failed. For example, the parameters passed from this <see cref="SerialPort"/>
         /// object were invalid.</exception>
-        /// <exception cref="ArgumentOutOfRangeException">The read time-out value is less than zero and not equal to <see cref="Timeout"/>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">The read time-out value is less than zero and not equal to <see cref="Timeout.Infinite"/>.</exception>
         public int ReadTimeout
         {
             get => _readTimeout;
 
             set
             {
-                if (value <= 0)
+                if (value is < 0 and value is not Timeout.Infinite)
```

Comment:
```suggestion
                if (value is < 0 and value is > Timeout.Infinite)
```

Reply:
Or... given this is probably an integer input... ` int.maxvalue` would probably be better?!

Reply:
```suggestion
               if (value is <Timeout.Infinite or value is > Int32.MaxValue)
```